app/models/grda_warehouse/client_file.rb

```diff
@@ -422,6 +424,24 @@ def sync_revokation_info(current_user)
       end
     end
 
+    def self.maintain_urls
+      where(url: nil).find_in_batches do |files|
+        batch = []
+        files.each do |file|
+          next unless file.client_file.attached?
+
+          batch << {
+            id: file.id,
+            url: file.client_file&.blob&.url,
+          }
+        rescue StandardError
+          # Ignore errors, this is less likely in production
+          # but if we can't find the url, we don't need to populate it
```

Comment:
Is there some way we could end up a with missing files? I imagine a missing file is not expected and indicates something has gone wrong. I imagine we wouldn't want to ignore errors

Reply:
We will only have missing files if ActiveStorage is unable to generate a url for the file (somehow the blob record was corrupted.)  I added this check since I have tons of terribly broken file data locally (from switching between file back-ends.)  
In production if we hit this, the file is essentially unusable, and for reporting and external data transfer would just be noise.

I'd love to find a more specific error to rescue, but was unable to given the errors I was seeing.

Reply:
I guess if this error were to occur in production, it would indicate an inconsistency between s3 and the db, which sounds bad; my understanding is that even if the object in s3 is garbage, it should still exist.

So if this is only for development, maybe we want:
```
raise unless Rails.env.development?
```

Or if we don't want to risk blocking the export:
```ruby
bad_file_ids = []
files.each do |file|
  #...
rescue
  bad_file_ids << file.id
end
Sentry.capture_message("Invalid Client Files: #{bad_file_ids.slice(0,100).inspect}") if bad_file_ids.any?
```